src/profiling/memory_meter.cpp

```diff
@@ -0,0 +1,64 @@
+#include <string>
+#include <vector>
+
+#include "memory_meter.hpp"
+#include <communication/global_policy.hpp>
+
+namespace nest {
+namespace mc {
+namespace util {
+
+namespace {
+    measurement collate(const std::vector<memory_size_type>& readings, std::string name) {
+        using gcom = communication::global_policy;
+
+        // Calculate the local change in allocated memory for each interval.
+        std::vector<memory_size_type> allocated;
+        allocated.push_back(0);
+        for (auto i=1u; i<readings.size(); ++i) {
+            allocated.push_back(readings[i] - readings[i-1]);
+        }
+
```

Comment:
I see your point. This block of code will be abstracted out in the following PR, and I will fix it there.